todoapp/src/components/Header.vue

```diff
@@ -1,16 +1,71 @@
 <!-- eslint-disable vue/multi-word-component-names -->
 <script setup lang="ts">
-import DeleteTasks from './DeleteTasks.vue'
+import plus from '../assets/plus.svg'
+import type { TextIdString } from '../types/text'
+import { computed } from 'vue'
+const props = defineProps<{
+  todos: TextIdString[]
+  modelValue: string
+}>()
 
-const emit = defineEmits(['deleteTasks'])
+const emit = defineEmits<{
+  (e: 'addTodo', todo: string): void
+  (e: 'update:modelValue', value: any): void
+}>()
 
-function deleteTasks() {
-  emit('deleteTasks')
+const inputValue = computed({
+  get() {
+    return props.modelValue
+  },
+  set(newValue) {
+    emit('update:modelValue', newValue)
+  }
+})
+
+function handleAdd() {
+  emit('addTodo', inputValue.value)
```

Comment:
You don't need to send this inputValue I think it's enough to just emit the addTodo action without any parameter